examples/dmd/pore_collapse.cpp

```diff
@@ -0,0 +1,299 @@
+    }
+
+    CAROM::DMD* dmd = nullptr;
+    CAROM::AdaptiveDMD* admd = nullptr;
+    if (dtc > 0.0){
+        dmd = new CAROM::DMD(dim, dtc);
+    }
+    else
+    {
+        dmd = new CAROM::AdaptiveDMD(dim, ddt, "LS", "G", dmd_epsilon);
+        admd = dynamic_cast<CAROM::AdaptiveDMD*> (dmd);
+    }
+
+    std::vector<std::string> training_par_list;
+    csv_db->getStringList(std::string(list_dir) + "/training_gpa", training_par_list, false);
+    int npar = training_par_list.size();
+
+    double* sample = new double[dim];
+    StopWatch dmd_training_timer, dmd_prediction_timer;
+    dmd_training_timer.Start();
+
+    for (int idx_test = 0; idx_test < npar; ++idx_test) 
+    {
+        std::string par_dir = training_par_list[idx_test]; // *gpa
+        if (myid == 0)
+        {
+            cout << "Loading samples for " << par_dir << " to train DMD." << endl;
+        }
+        std::vector<std::string> snap_list;
+        csv_db->getStringList(std::string(list_dir) + "/" + par_dir, snap_list, false);
+        int num_snap = snap_list.size();
+        double tval = 0.0;
+        for (int idx_snap = 0; idx_snap < num_snap; ++idx_snap)
+        {
+            std::string snap = snap_list[idx_snap]; // run_036.*
+            csv_db->getDoubleArray(std::string(data_dir) + "/" + par_dir + "/" + snap + "/tval.txt", &tval, 1);
+            if (idx_snap == 0)
+            {
+                t_init = tval;
+            }
+            std::string data_filename = std::string(data_dir) + "/" + par_dir + "/" + snap + "/" + variable + ".csv"; // {zone_*,tkelv}.csv
+            csv_db->getDoubleArray(data_filename, sample, nelements, idx_state);
+            dmd->takeSample(sample, tval - t_init);
+        }
+        if (myid == 0)
+        {
+            cout << "Loaded " << num_snap << " samples for " << par_dir << "." << endl;
+        }
+    }
+
+    if (rdim != -1)
+    {
+        if (myid == 0)
+        {
+            cout << "Creating DMD with rdim: " << rdim << endl;
+        }
+        dmd->train(rdim);
+    }
+    else if (ef != -1)
+    {
+        if (myid == 0)
+        {
+            cout << "Creating DMD with energy fraction: " << ef << endl;
+        }
+        dmd->train(ef);
+    }
+
+    dmd_training_timer.Stop();
+
+    CAROM::Vector* result = new CAROM::Vector(dim, true);
+    if (admd)
+    {
+        MFEM_VERIFY(npar == 1, "Adaptive DMD only works with 1 training parameter.");
+        dtc = admd->getTruedt();
+        const CAROM::Matrix* f_snapshots = admd->getInterpolatedSnapshots();
+        CAROM::Vector* isnap = new CAROM::Vector(dim, true);
+        if (myid == 0)
+        {
+            cout << "Verifying Adaptive DMD solution." << endl;
+        }
+        for (int k = 0; k < f_snapshots->numColumns(); ++k)
+        {
+            double tval = t_init + k*dtc;
+            f_snapshots->getColumn(k, isnap);
+            result = admd->predict(tval-t_init);
```

Comment:
I implemented a time offset in AdaptiveDMD. I believe this can be changed to just "tval". d_t_offset in AdaptiveDMD will be set to t_init and will do the subtraction for yo.